### DISCUSSION TITLE
Detect current node according to the cursor position

### DISCUSSION THREAD

**Discussion by uyha** [L3MON4D3/LuaSnip#967] [Q&A]
Is there a way to get the node (assuming there is one) which the cursor is inside it?

**Comment by L3MON4D3** [comment]
Oh, that will tangentially be included in #941!
What do you want to use it for?

**Reply by uyha** [reply]
just following https://github.com/L3MON4D3/LuaSnip/discussions/963, trying make it possible to expand the current snippet instead of the last snippet.

**Reply by L3MON4D3** [reply]
Not sure what you mean by that, could you elaborate?

**Reply by uyha** [reply]
so following what I have in #963, the code will only work for the last dynamic node and not the node that the cursor is currently in. This is because in the function, `ls.session.current_nodes` is used, and AFAIK, it only returns the last used node (please correct me if i’m wrong). Hence, I want to know if there’s a way to detect the node the cursor is currently at.

**Reply by L3MON4D3** [reply]
Ahhh okay.
Yeah, it should be possible to just activate the node at the cursor by querying, and then focusing it :+1:
I'll think about some api to do that.

Just to make sure I got everything correctly, `detect node at cursor -> make node at cursor the current_node -> trigger dynamicNode` works for you? (I guess we could also, technically, drop the second step, but it would probably need some glue-code in your config (not much))

**Reply by uyha** [reply]
I think an API for getting the node at the cursor is enough, since step 2 is not needed as far as I can tell, and step 3 can be done manually anyway.
